mlflow/gateway/utils.py

```diff
@@ -38,3 +56,97 @@ def kill_child_processes(parent_pid):
     _, still_alive = psutil.wait_procs(parent.children(), timeout=3)
     for p in still_alive:
         p.kill()
+
+
+def _is_valid_uri(uri: str):
+    """
+    Evaluates the basic structure of a provided gateway uri to determine if the scheme and
+    netloc are provided
+    """
+    try:
+        parsed = urlparse(uri)
+        return all([parsed.scheme, parsed.netloc])
+    except ValueError:
+        return False
+
+
+def _is_gateway_server_available(gateway_uri: str):
+    server_health_url = urljoin(gateway_uri, MLFLOW_GATEWAY_HEALTH_ENDPOINT)
+
+    try:
+        response = _get_gateway_response_with_retries("GET", server_health_url)
+        augmented_raise_for_status(response)
+    except HTTPError as http_err:
+        _logger.warning(f"There is not a gateway server running at {gateway_uri}. {http_err}")
+        return False
+    except Exception as err:
+        _logger.warning(
+            f"Unable to verify if a gateway server is healthy at {gateway_uri}. Error: {err}"
+        )
+        return False
+    else:
+        return True
+
+
+def _merge_uri_paths(paths: List[str]) -> str:
+    sanitized = [part.strip("/") for part in paths]
+    return "/".join(sanitized)
+
+
+def _resolve_gateway_uri(gateway_uri: Optional[str] = None) -> str:
+    gateway = gateway_uri or get_gateway_uri()
+    return gateway
```

Comment:
```suggestion
    return gateway_uri or get_gateway_uri()
```

Reply:
Can we use an immediate return?

- To avoid useless assignments
- To remove useless lines
- To simplify functions

Reply:
Example where an immediate return makes sense.

```python
# can't tell what this function does until I reach the final line

def f():
    a = {
        # 30 lines
        ...
    }
    return a
```


```python
# can immediately tell what this function does

def f():
    return {
        # 30 lines
        ...
    }
```

Reply:
I was going to put an exception check in there for validating the gateway definition, but I think I'll leave it in the higher level `get_gateway_uri()`. I'll simplify the return